bin/xud

```diff
@@ -87,6 +87,16 @@ const { argv } = require('yargs')
       type: 'boolean',
       default: undefined,
     },
+    'p2p.discover': {
+      describe: 'Do not send GET_NODES if false, when connecting to a peer',
+      type: 'boolean',
+      default: false,
```

Comment:
The way defaults arguments work from the command line is they overwrite both the hardcoded defaults and what's in the config file. The default for boolean types should be `undefined` to avoid this behavior.

Reply:
Oh didn't notice this, thanks

Reply:
@sangaman can you please explain why we need `undefined` here? what problem is this solving?

Reply:
Yeah it has to do with how yargs works. All booleans in yargs have a "default" default value of false - as in if you don't manually specify a default value, it will pass false for that arguent if `type` is `boolean`. The way our config works, command line arguments override the hardcoded defaults and what's in the config file. If we didn't have an `undefined` default here, `p2p.discover` would always be `false` unless we set `--p2p.discover true` from the command line.